Makefile

```diff
@@ -1,4 +1,18 @@
 # ---------------------------------------------------------------------
+# Build configuration
+# ---------------------------------------------------------------------
+
+BUILD_DIRECTORY ?= release
+
+# See http://stackoverflow.com/a/20763842/1641422
+ifeq ($(wildcard $(dir $(BUILD_DIRECTORY)).),)
+$(error $(dir $(BUILD_DIRECTORY)) does not exist)
```

Comment:
These two lines _might_ be slightly more readable if `$(dir $(BUILD_DIRECTORY))` is factored out to e.g. a separate `BUILD_DIRECTORY_PARENT` variable?